feat(IT Wallet): [SIW-2430] Add thank you screen after proximity presentation

adelloste opened the pull request:
This PR depends on https://github.com/pagopa/io-app/pull/7086

## Short description
This PR adds a thank you screen, which is shown at the end of the presentation flow if everything completes successfully

## List of changes proposed in this pull request
- Added the `ItwProximitySendDocumentsResponseScreen`
- Updated the state machine to handle success state transition
- Improved error handling

## Demo

<details open><summary>Proximity</summary>
<p>

| Proximity |
|--------|
| <video src="https://github.com/user-attachments/assets/cb00e0f2-29cb-41ee-996d-11ab0b0f18aa"></video> |

</p>
</details>

## How to test
> [!NOTE]  
> This PR cannot be fully tested at the moment because we don't currently have an available mDL credential.
> However, if you want to try the flow, you can do by mocking both:
> - the value returned by [isItwCredential](https://github.com/pagopa/io-app/blob/5f95677e94fb50a155b0879cd993884adc0bb283/ts/features/itwallet/common/utils/itwCredentialUtils.ts#L101)
> - the mDL credential itself
> 
> To mock the mDL credential, update the [credentials](https://github.com/pagopa/io-app/blob/5f95677e94fb50a155b0879cd993884adc0bb283/ts/features/itwallet/presentation/proximity/machine/actions.ts#L23) by injecting the [mock](https://github.com/pagopa/io-app/blob/5f95677e94fb50a155b0879cd993884adc0bb283/ts/features/itwallet/presentation/proximity/__mocks__/credentials.ts#L4) as shown below:
> ```
> return {
>   credentialsByType: pipe(
>     {
>       ...credentialsByType,
>       ...MOCK_CREDENTIALS
>     },
>     R.filterMap(identity)
>   )
> };
> ```
> **Important:**
> Make sure to update the `keyTag` inside `MOCK_CREDENTIALS`.
> It must match the `keyTag` of the credential that is persisted in the store.
> Once the mocks are in place, you can follow the steps below.

- navigate to the credential details
- tap the "Mostra QR di verifica" button
- scan the QR code using the verifier app
- proceed with the proximity flow
- after authorizing the data presentation via PIN/biometric authentication, if everything goes well, you should see the thank you screen

mastro993 commented on ts/features/itwallet/presentation/proximity/screens/ItwProximityClaimsDisclosureScreen.tsx:
```diff
@@ -44,19 +43,11 @@ export const ItwProximityClaimsDisclosureScreen = () => {
       () => (
-        <LoadingScreenContent
-          contentTitle={I18n.t(
+        <ItwProximityLoadingScreen
```
Isn't this the same as `LoadingScreenContent`?
If yes, do we really need a custom component like `ItwProximityLoadingScreen` just to handle one specific case?

adelloste replied:
Addressed in https://github.com/pagopa/io-app/pull/7101/commits/635c11de134d3dcea0d4e84cf6412337830baa95 since it was redundant

adelloste commented:
> I encountered an issue when testing on a Samsung A34 with an iPhone 11 Pro as the verifier app: the onDeviceConnecting event is not received, so the machine does not navigate to the claims disclosure screen and it is not possible to continue the presentation.

You're right, the issue occurs because the current Android SDK doesn't emit the `onDeviceConnecting` event. To fix this, I've moved the navigation when the `onDeviceConnected` event is received.
Addressed in https://github.com/pagopa/io-app/pull/7101/commits/7195eeb38b9fe50f949186ef9920cd356927f5d9

mastro993 reviewed (approved):
With the correct mocks it works smoothly! LGTM
